### PR TITLE
Implement `picks` argument to `Raw.plot()`

### PR DESCRIPTION
**Pull request by ivopascal**
Implements #12355 

An arguments `picks` is added to `Raw.plot()` to offer it the same functionality as `Epochs.plot()` and `Evoked.plot()`.

Picks selects a subset of channels to be visualised, but ignores the order of them. `order=` does specify the sequence. Any interactions are resolved in the same way as `Epochs.plot()`.

### CODE REVIEW

**Comment by ivopascal** [comment]
`No matching distribution found for numpy>=2.0.0.dev0`

I think https://anaconda.org/scientific-python-nightly-wheels/ is experiencing some hick-ups, as nothing from `numpy` is showing there. The pipelines of other PRs were able to pull from there just fine 12 hours ago.

**Comment by ivopascal** [comment]
Ah, it's a known problem https://github.com/numpy/numpy/issues/25907

**Comment by matthewfeickert** [comment]
@ivopascal Please retrigger the CI given https://github.com/numpy/numpy/issues/25907#issuecomment-1973431825.

**Comment by larsoner** [comment]
We will once https://github.com/numpy/numpy/issues/25907#issuecomment-1973484113 is fixed :)

File: doc/changes/devel/12467.newfeature.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Add ``picks=`` to :meth:`mne.io.Raw.plot` to improve uniformity with :meth:`mne.Epochs.plot` and :meth:`mne.Evoked.plot`, by `Ivo de Jong`_.
```
Comment (hoechenberger) [inline doc/changes/devel/12467.newfeature.rst:1]:
```suggestion
Add ``picks`` parameter to :meth:`mne.io.Raw.plot`, allowing users to select which channels to plot. This makes makes the raw data plotting API consistent with :meth:`mne.Epochs.plot` and :meth:`mne.Evoked.plot`, by `Ivo de Jong`_.
```

File: doc/changes/devel/12467.newfeature.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Add ``picks`` parameter to :meth:`mne.io.Raw.plot`, allowing users to select which channels to plot. This makes makes the raw data plotting API consistent with :meth:`mne.Epochs.plot` and :meth:`mne.Evoked.plot`, by `Ivo de Jong`_.
```
Comment (larsoner) [inline doc/changes/devel/12467.newfeature.rst:1]:
```suggestion
Add ``picks`` parameter to :meth:`mne.io.Raw.plot`, allowing users to select which channels to plot. This makes makes the raw data plotting API consistent with :meth:`mne.Epochs.plot` and :meth:`mne.Evoked.plot`, by :newcontrib:`Ivo de Jong`.
```

**Pull request merged**